### ISSUE TITLE
Support ions in MBPol Three Body Force, reference platform

### ISSUE THREAD

**Issue by zonca** [paesanilab/mbpol_openmm_plugin#74]
@pushpbajaj , @sqamara is ready to start on the ThreeBody Force,

can you please post on a gist:

* computation of the force
* polynomial

and here as a comment on the thread:

* energy and forces for a 2 waters 1 Cl system (possibly get the first 2 water molecules of the usual water3 system.

Thanks

TODO:

- [ ] implement unit test with 3 waters and 1 Cl

**Comment by pushpbajaj** [comment]
@zonca @sqamara Here's a gist with h2o-cl 3B polynomials and the code which computes the 3B polynomials and the forces

https://gist.github.com/pushpbajaj/2b9fce0b6bdee21c76a1

I will upload a test case with the 3B forces

**Comment by pushpbajaj** [comment]
@zonca  @sqamara Here's a test for the 3B polynomial part 

For this configuration: 
```
O1          0.2071123419        0.1431350648        0.2575948809
H1          0.3968154394       -0.4466985822       -0.5164780569
H1         -0.0790971024        0.9461358941       -0.1868734592
O2          3.1469464639        0.2657051814       -0.5347590767
H2          2.3878654247        0.3762497422        0.0532565299
H2          2.7252172819       -0.2612759028       -1.2421153352
Cl          1.0072476456       -1.3374671605       -2.3154322027
```
which is the same as the other 2water-cl configuration I used for the "total 2B energy" test.

the 3B polynomial energy =  1.912840868e-01 kcal/mol

3B polynomial gradients
```
Atom 1   4.507870220e-01   -6.667749758e+00   -3.254399711e+00
Atom 2   1.759702035e+00   -1.837574575e+00    2.740586272e+00
Atom 3  -1.807260132e+00    7.162924177e+00   -5.635704934e-01
Atom 4   7.451693208e+00   -1.688792853e+00   -7.490080448e+00
Atom 5  -6.832490586e+00    3.254958361e+00    8.077659805e+00
Atom 6   3.401858119e-01   -4.915100229e-01    7.333484663e-01
Atom 7  -1.362617360e+00    2.677446706e-01   -2.435438912e-01
```

**Comment by sqamara** [comment]
@zonca I've integrated the h2o-cl code into the MBPolReferenceThreeBodyForce and it passes the test. here is the pull request https://github.com/paesanilab/mbpol_openmm_plugin/pull/75

**Comment by zonca** [comment]
good job @sqamara , 

@pushpbajaj I think it is time to prepare a more difficult test to pass.
It would be great if you can create a test with 3 water molecules (from the usual water3 system) and 1 Chloride. So we can check if @sqamara 's implementation is correctly identifying water-water-water and water-water-ion triplets.

**Comment by pushpbajaj** [comment]
@sqamara @zonca 

here is a test case with three water molecules and one chloride. 

For this configuration: 
```
 O          1.5227851364       -0.7997883006       -1.1599636805
 H          1.3222088042       -0.5997860560       -0.2220811889
 H          0.6598559742       -1.1386138839       -1.4424152319
 O         -0.0644190117        1.7383223248       -1.1597784472
 H         -0.1416949835        1.4655959044       -0.2224117958
 H          0.6566082745        1.1546166383       -1.4406133606
 O         -1.4714724790       -0.9089637376       -1.1593267351
 H         -1.1951877835       -0.8374906804       -0.2222208880
 H         -1.3301291268        0.0071609948       -1.4418538065
 Cl        -0.0039350888        0.0039367723        1.5059305306 
```

The total 3B wat-wat-cl polynomial energy is 0.6087545 kcal/mol
The wat-wat-wat 3B polynomial energy is -0.0991616 kcal/mol 

total gradients only from 3B wat-wat-cl polynomials: 
```
Atom 1      1.219679470e+01   5.682012278e+00   9.469287405e+00
Atom 2      3.660672115e+00   4.215018098e-01   2.086944346e+00
Atom 3     -1.487241456e+01  -6.403786450e+00  -1.146885060e+01
Atom 4     -1.098508231e+01   7.818413472e+00   9.347566284e+00
Atom 5     -2.166415570e+00   2.960253056e+00   2.169006450e+00
Atom 6      1.293656393e+01  -9.779489082e+00  -1.143839990e+01
Atom 7     -1.206889286e+00  -1.337857660e+01   9.483751927e+00
Atom 8     -1.482615503e+00  -3.379218375e+00   2.071696153e+00
Atom 9      1.933667385e+00   1.606070854e+01  -1.145365654e+01
Atom 10    -1.428089532e-02  -1.818655121e-03  -2.673455204e-01
```

**Comment by zonca** [comment]
great, thanks @pushpbajaj

**Comment by zonca** [comment]
last test implemented in b6e9769.

Implementation completed, @zonca to review #77 and merge.
